File: $_algotsu-kaizen/problemo-set/xor-queries-of-a-subarray.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

 int* xorQueries(int* arr, int arrSize, int** queries, int queriesSize, int* queriesColSize, int* returnSize) {
    int *result = (int *)malloc(sizeof(int) * queriesSize);
    int *prexor = (int *)malloc(sizeof(int) * arrSize);
    int xor = 0b0;
    for (int i = 0; i < arrSize; ++i) {
        xor ^= *(arr + i);
        *(prexor + i) = xor;
    }

    for (int i = 0; i < queriesSize; ++i) {
        int l = **(queries + i), r = *(*(queries + i) + 1);
        *(result + i) =  (l > 0 ? *(prexor + l - 1) : 0 ) ^ *(prexor + r);
    }

    free(prexor);
    *returnSize = queriesSize;
    return result;
}

File: $_algotsu-kaizen/problemo-set/xor-queries-of-a-subarray.rs
impl Solution {

    pub fn xor_queries(arr: Vec<i32>, queries: Vec<Vec<i32>>) -> Vec<i32> {
        let mut xor: i32 = 0b0;
        let mut prexor: Vec<i32> = Vec::new();
        let mut result: Vec<i32> = Vec::new();
        let mut idx: usize = 0;

        for e in arr {
            xor = xor ^ e;
            prexor.push(xor);
        }

        for querie in queries {
            let l = querie[0] as usize;
            let r = querie[1] as usize;
            result.push((if l > 0 { prexor[l - 1] } else { 0 } ) ^ prexor[r]);
        }

        result
    }
}
